src/timer-utils.js

```diff
@@ -0,0 +1,36 @@
+export const DEFAULT_RETRY_CONFIG = {
+    interval: { ms: 0 },
+    maxAttempts: 1
+}
+
+/**
+ * A generic retry mechanism for synchronous functions that return
+ * a 'success' or 'error' response
+ *
+ * @template T
+ * @template {{ success: T } | { error: { message: string } }} FnReturn
+ * @param {() => FnReturn} fn
+ * @param {typeof DEFAULT_RETRY_CONFIG} [config]
+ * @return {Promise<{ result: FnReturn | undefined, exceptions: string[] }>}
+ */
+export async function retry (fn, config = DEFAULT_RETRY_CONFIG) {
+    let lastResult
+    const exceptions = []
+    for (let i = 0; i < config.maxAttempts; i++) {
+        try {
+            lastResult = fn()
+        } catch (e) {
+            exceptions.push(e.toString())
+        }
+
+        // stop when there's a good result to return
+        if (lastResult && 'success' in lastResult) break
```

Comment:
Maybe a silly question, but where do we get 'success' from?

Reply:
Added comments :)